main.go

```diff
@@ -63,12 +62,41 @@ type SignatureData struct {
 	Payloads map[string]string `json:"Payloads"`
 }
 
+var configFilePath = flag.String("config-file", "", "path to the config file with target image digest, root of trust, signature, and verifier data")
+
+var proxyURL = flag.String("proxy-url", "", "")
+var proxyHasCredentials = flag.Bool("proxy-has-credentials", false, "")
+
+func getProxyDetails() (Proxy, error) {
+	proxy := Proxy{URL: *proxyURL}
+	if *proxyURL != "" && *proxyHasCredentials {
+		fmt.Println("getting proxy credential details")
+		stdinCredentials, err := os.ReadFile(os.Stdin.Name())
+		if err != nil {
+			return Proxy{}, fmt.Errorf("error when reading credentials, could not read stdin: %s", err.Error())
+		}
+		splitCredentials := strings.Split(string(stdinCredentials), ":")
+		if len(splitCredentials) != 2 {
+			return Proxy{}, fmt.Errorf("error when reading credentials, invalid format, credential string should have single colon (USERNAME:PASSWORD)")
```

Comment:
why proxy password cannot contains ':' ?

Reply:
I will change this to behave like the URL specification: username cannot contain colon, password can contain colon.